File: paddler_client_python/paddler_client/balancer_desired_state.py
from pydantic import BaseModel, Field

from paddler_client.agent_desired_model import AgentDesiredModel
from paddler_client.chat_template import ChatTemplate
from paddler_client.inference_parameters import InferenceParameters


class BalancerDesiredState(BaseModel):
    chat_template_override: ChatTemplate | None = None
    inference_parameters: InferenceParameters = Field(
        default_factory=InferenceParameters,
    )
    model: AgentDesiredModel = Field(default_factory=AgentDesiredModel.none)
    multimodal_projection: AgentDesiredModel = Field(
        default_factory=AgentDesiredModel.none,
    )
    use_chat_template_override: bool = False

File: paddler_client_javascript/src/schemas/BalancerDesiredState.ts
import { z } from "zod";

import { AgentDesiredModelSchema } from "./AgentDesiredModel";
import { ChatTemplateSchema } from "./ChatTemplate";
import { InferenceParametersSchema } from "./InferenceParameters";

export const BalancerDesiredStateSchema = z
  .object({
    chat_template_override: ChatTemplateSchema.nullable(),
    inference_parameters: InferenceParametersSchema,
    model: AgentDesiredModelSchema,
    multimodal_projection: AgentDesiredModelSchema,
    use_chat_template_override: z.boolean(),
  })
  .strict();

export type BalancerDesiredState = z.infer<typeof BalancerDesiredStateSchema>;
